File: jianzhioffer/firstround/jianzhi_57.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if not nums or len(nums) == 1:
            return []
        left, right = 0, len(nums)-1
        while left < right:
            sum = nums[left] + nums[right]
            if sum < target:
                left += 1
            elif sum > target:
                right -= 1
            else:
                return [nums[left], nums[right]]
        return []


if __name__ == '__main__':
    n = [2,7,11,15]
    target = 18
    obj = Solution()
    result = obj.twoSum(n, target)
    print(result)

File: jianzhioffer/secondround/jianzhi_57.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution
{
public:
    Solution() {};
    ~Solution() {};
    vector<int> twoSum(vector<int>& nums, int target);
};

vector<int> Solution::twoSum(vector<int>& nums, int target)
{
    vector<int> res{};
    if(nums.empty() || target<=0) return res;
    int left=0, right=nums.size()-1;
    while (left<right)
    {
        if(nums[left]+nums[right]>target) right--;
        else if(nums[left]+nums[right]<target) left++;
        else
        {
            res = {nums[left], nums[right]};
            break;
        }
    }
    return res;
}

int main(int argc, char const *argv[])
{
    vector<int> nums = {1,4,5,6,8,10};
    Solution obj;
    vector<int> result = obj.twoSum(nums, 10);
    return 0;
}
